vendor/bat-native-ads/src/bat/ads/internal/federated/covariate_logs.cc

```diff
@@ -18,18 +19,149 @@
 #include "bat/ads/internal/federated/log_entries/ad_notification_impression_served_at_covariate_log_entry.h"
 #include "bat/ads/internal/federated/log_entries/ad_notification_locale_country_at_time_of_serving_covariate_log_entry.h"
 #include "bat/ads/internal/federated/log_entries/ad_notification_number_of_tabs_opened_in_past_30_minutes_log_entry.h"
+#include "bat/ads/internal/federated/log_entries/average_clickthrough_rate.h"
+#include "bat/ads/internal/federated/log_entries/last_ad_notification_was_clicked.h"
+#include "bat/ads/internal/federated/log_entries/number_of_user_activity_events.h"
+#include "bat/ads/internal/federated/log_entries/time_since_last_user_activity_event.h"
+#include "bat/ads/internal/logging.h"
 
 namespace ads {
 
+namespace {
+
+CovariateLogs* g_covariate_logs = nullptr;
+
+static base::NoDestructor<
```

Comment:
The pattern is to put static into a function. In this case it will be created on demand. Now it is created on ads library process start. We could change this as follow up

Reply:
sounds good, I'll create ticket to fix